pkg/cloudfoundry/ManifestUtils.go

```diff
@@ -0,0 +1,226 @@
+	}
+
+	err = yaml.Unmarshal(content, &m.self)
+
+	if err != nil {
+		return m, errors.Wrapf(err, "Cannot parse yaml file '%s': %s", m.name, string(content))
+	}
+
+	log.Entry().Infof("Manifest file '%s' has been parsed", m.name)
+
+	return m, nil
+}
+
+// WriteManifest ...
+func (m *Manifest) WriteManifest() error {
+
+	d, err := yaml.Marshal(&m.self)
+	if err != nil {
+		return err
+	}
+
+	log.Entry().Debugf("Writing manifest file '%s'", m.name)
+	err = ioutil.WriteFile(m.name, d, 0644)
+
+	if err == nil {
+		m.modified = false
+	}
+
+	log.Entry().Debugf("Manifest file '%s' has been written", m.name)
+	return err
+}
+
+// GetName ...
+func (m Manifest) GetName() string {
+	return m.name
+}
+
+// GetApplications ...
+func (m Manifest) GetApplications() ([]interface{}, error) {
+	return toSlice(m.self)
+}
+
+// ApplicationHasProperty Checks if the application denoted by 'index' has the property 'name'
+func (m Manifest) ApplicationHasProperty(index int, name string) (bool, error) {
+
+	sliced, err := toSlice(m.self[propApplications])
+
+	if err != nil {
+		return false, err
+	}
+
+	if index >= len(sliced) {
+		return false, fmt.Errorf("Index (%d) out of bound. Number of apps: %d", index, len(sliced))
+	}
+
+	_m, err := toMap(sliced[index])
+
+	if err != nil {
+		return false, err
+	}
+
+	_, ok := _m[name]
+
+	return ok, nil
+}
+
+// GetApplicationProperty ...
+func (m Manifest) GetApplicationProperty(index int, name string) (interface{}, error) {
+
+	sliced, err := toSlice(m.self[propApplications])
+
+	if err != nil {
+		return "", err
+	}
+
+	if index >= len(sliced) {
+		return "", fmt.Errorf("Index (%d) out of bound. Number of apps: %d", index, len(sliced))
+	}
+
+	app, err := toMap(sliced[index])
+
+	if err != nil {
+		return "", err
+	}
+
+	if app[name] != nil {
+		return app[name], nil
+	}
+
+	return "", fmt.Errorf("No such property: '%s' available in application at position %d", name, index)
+}
+
+// GetAppName Gets the name of the app at 'index'
+func (m Manifest) GetAppName(index int) (string, error) {
+
+	appName, err := m.GetApplicationProperty(index, "name")
+
+	if err != nil {
+		return "", err
+	}
+
+	if name, ok := appName.(string); ok {
+		return name, nil
+	}
+
+	return "", fmt.Errorf("Cannot retrieve application name for app at index %d", index)
+}
+
+// Transform For each app in the manifest the first entry in the build packs list
+// gets moved to the top level under the key 'buildpack'. The 'buildpacks' list is
+// deleted.
+func (m *Manifest) Transform() error {
+
+	sliced, err := toSlice(m.self[propApplications])
+	if err != nil {
+		return err
+	}
+
+	for _, app := range sliced {
+		appAsMap, err := toMap(app)
+		if err != nil {
+			return err
+		}
+
+		err = transformApp(appAsMap, m)
+
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func transformApp(app map[string]interface{}, m *Manifest) error {
+
+	appName := "n/a"
+
+	if n, ok := app["name"].(string); ok {
+		if len(n) > 0 {
```

Comment:
Why not: `if n != "" {`?

Reply:
Would be also ok. But why not `len(n) > 0`?

Reply:
I was just curious, keep it the way it is. My thought was: `len()` is an extra method call while comparing to `""` saves you that : }

Reply:
If some code does or does not result in extra method calls is probably only definitely visible in the generated assembly. Both `len()` and `==` are probably built-ins and likely end up be optimized to the exact same machine code. I mean, just because `len()` looks like a function call, doesn't mean `==` is not a function call.

Reply:
Yeah, I believe you are right. But just to be clear, what I mean is that `len() >` looks like two function calls, while `==` looks like one two me.